test/headful.spec.js

```diff
@@ -85,26 +85,33 @@ describeChromeOnly('headful tests', function() {
       expect(pages).toEqual(['about:blank']);
       await browser.close();
     });
-    it('headless should be able to read cookies written by headful', async() => {
-      const { server, puppeteer } = getTestState();
-
-      const userDataDir = await mkdtempAsync(TMP_FOLDER);
-      // Write a cookie in headful chrome
-      const headfulBrowser = await puppeteer.launch(Object.assign({userDataDir}, headfulOptions));
-      const headfulPage = await headfulBrowser.newPage();
-      await headfulPage.goto(server.EMPTY_PAGE);
-      await headfulPage.evaluate(() => document.cookie = 'foo=true; expires=Fri, 31 Dec 9999 23:59:59 GMT');
-      await headfulBrowser.close();
-      // Read the cookie from headless chrome
-      const headlessBrowser = await puppeteer.launch(Object.assign({userDataDir}, headlessOptions));
-      const headlessPage = await headlessBrowser.newPage();
-      await headlessPage.goto(server.EMPTY_PAGE);
-      const cookie = await headlessPage.evaluate(() => document.cookie);
-      await headlessBrowser.close();
-      // This might throw. See https://github.com/puppeteer/puppeteer/issues/2778
-      await rmAsync(userDataDir).catch(e => {});
-      expect(cookie).toBe('foo=true');
-    });
+    itFailsWindowsUntilDate(
+        /* We have deferred fixing this test on Windows in favour of
+         * getting all other Windows tests running on CI. Putting this
+         * date in to force us to come back and debug properly in the
+         * future.
+         */
+        new Date(2020, 5, 1),
```

Comment:
Readability (to avoid the off-by-one-month gotcha):

```suggestion
        new Date('2020-06-01'),
```